inc/packages/namespace.php

```diff
@@ -0,0 +1,580 @@
+	// Get the package ID to pass to action hook.
+	$package_id = get_package_id( $slug, $for_network );
+	if ( empty( $package_id ) ) {
+		return new WP_Error(
+			'delete_package_error',
+			// translators: %s replaced by package slug.
+			sprintf( __( 'Package for slug %s does not exist', 'altis' ), $slug )
+		);
+	}
+
+	$file = get_package_path( $slug, $for_network );
+
+	// Delete the file.
+	unlink( $file );
+
+	// Remove the stored package ID.
+	if ( $for_network ) {
+		$deleted = delete_site_option( "altis_search_package_{$slug}" );
+		delete_site_option( "altis_search_package_status_{$slug}" );
+		delete_site_option( "altis_search_package_error_{$slug}" );
+	} else {
+		$deleted = delete_option( "altis_search_package_{$slug}" );
+		delete_option( "altis_search_package_status_{$slug}" );
+		delete_option( "altis_search_package_error_{$slug}" );
+	}
+
+	// Update the index settings on successful deletion.
+	if ( $deleted ) {
+		do_settings_update( $for_network, strpos( $slug, 'user-dictionary' ) !== false );
+
+		/**
+		 * Action triggered when a package is deleted.
+		 *
+		 * @param string $package_id The package ID.
+		 * @param string $slug The package slug.
+		 * @param bool $for_network Whether this is for the network level or site level.
+		 */
+		do_action( 'altis.search.deleted_package', $package_id, $slug, $for_network );
+
+		return true;
+	}
+
+	return new WP_Error(
+		'delete_package_error',
+		// translators: %s replaced by package slug.
+		sprintf( __( 'Failed to delete package with slug %s', 'altis' ), $slug )
+	);
+}
+
+/**
+ * Return a comma separated list of all default indexables for the given site.
+ *
+ * @param integer|null $blog_id The site to get index names for. Defaults to current site.
+ * @return string
+ */
+function get_site_indices( ?int $blog_id = null ) : string {
+	$indexes = [];
+
+	// Check for default indexes and their existence.
+	foreach ( Indexables::factory()->get_all() as $indexable ) {
+		if ( ! $indexable->index_exists( $blog_id ) ) {
+			continue;
+		}
+		$indexes[] = $indexable->get_index_name( $blog_id );
+	}
+
+	return implode( ',', $indexes );
+}
+
+/**
+ * Update the index settings after upload. For stopwords and synonyms
+ * no reindexing of data is required as they are applied at search time.
+ *
+ * If the data needs to be updated such as if the Japanese user dictionary
+ * is updated then $update_index allows for updating using `_update_by_query`.
+ *
+ * Note this function should be run as a background task only.
+ *
+ * @param boolean $for_network Whether this is a network wide mapping update.
+ * @param boolean $update_data Whether the index data should be updated as well.
+ * @return void
+ */
+function do_settings_update( bool $for_network = false, bool $update_data = false ) : void {
+	// Get latest settings.
+	$mapping = Enhanced_Search\elasticpress_mapping( [] );
+	$settings = $mapping['settings'];
+
+	if ( $for_network ) {
+		$sites = Utils\get_sites( 0 );
```

Comment:
Spicy. What does 0 mean in this context?

Reply:
That's just how ElasticPress does it - I think it refers to the network, 0 meaning current network because reasons.